rust/kernel/src/lib.rs

```diff
@@ -56,3 +58,24 @@ fn panic(_info: &PanicInfo) -> ! {
 
 #[global_allocator]
 static ALLOCATOR: allocator::KernelAllocator = allocator::KernelAllocator;
+
+/// Attempts to allocate memory for the given value using the global allocator. On success, the
+/// value is moved into it and returned to the caller wrapped in a Box.
+pub fn try_alloc<T>(value: T) -> KernelResult<Box<T>> {
+    let layout = core::alloc::Layout::for_value(&value);
+    let rptr = unsafe { alloc::alloc::alloc(layout) };
+    if rptr.is_null() {
+        return Err(Error::ENOMEM);
+    }
+    let ptr = rptr as *mut T;
+    unsafe {
+        ptr.write(value);
+        Ok(Box::from_raw(ptr))
+    }
+}
+
+/// Attempts to allocate memory for the given value using the global allocator. On success, the
+/// value is moved into it and returned to the caller wrapped in a pinned Box.
```

Comment:
`` `value` `` and `` `Box` ``. Same for `try_alloc`

Reply:
Done.